src/file_utils.rs

```diff
@@ -104,3 +104,59 @@ pub(crate) fn decode_resource_data_entry(yaml_location: &YamlLocation, value_at_
     }
     .clone())
 }
+
+pub(crate) fn add_annotation(resource: &mut Value, yaml_location: &YamlLocation) -> Result<()> {
```

Comment:
I find this function to be hard to understand without background / an example

```suggestion
/// Adds annotation to kubernetes resources to indicate that they have been edited by recert. The
/// annotation has a string value that is a serialized json object with its keys being JSON
/// pointers of fields in this resource edited by recert and values being a list of PEM bundle
/// indices of cryptographic objects that were edited in that field (or "N/A" if the field was not
/// PEM-encoded).
///
/// For example:
///
/// metadata:
///   annotations:
///     ...
///     recert-edited: '{"/spec/config/storage/files/15/contents/source":["1","2","6","5","3","0","4"],"/spec/config/storage/files/21/contents/source":["0"]}'
///     ...
///
/// This annotation is purely informational and is not used programmatically by recert itself. It's
/// used to help troubleshooters notice this is not a "natural" cryptographic object, but instead
/// one that was manipulated by recert.
pub(crate) fn add_annotation(resource: &mut Value, yaml_location: &YamlLocation) -> Result<()> {
```